### PR TITLE
DOC: `distance_transform_bf` example

### PR DESCRIPTION
**Pull request by dschmitz89**
#### Reference issue
#7168

#### What does this implement/fix?
Adds examples for `ndimage.distance_transform_bf`.

#### Additional information
Admittedly, the example might look more like a tutorial in image plotting with matplotlib at the moment. Especially, more explanations could be added. Suggestions welcome, this is also my first PR for `image`.

### CODE REVIEW

File: scipy/ndimage/_morphology.py [hunk 1]
```diff
@@ -1922,12 +1922,93 @@ def distance_transform_bf(input, metric="euclidean", sampling=None,
...
+    >>> image = np.zeros((100, 100), dtype=np.uint8)
+    >>> image[35:65, 20:80] = 1
+    >>> image = add_circle(28, 65, 10, image)
+    >>> image = add_circle(37, 30, 10, image)
+    >>> image = add_circle(70, 45, 20, image)
+    >>> image = add_circle(45, 80, 10, image)
+    >>> fig, ax = plt.subplots()
+    >>> ax.imshow(image, cmap='gray')
+    >>> ax.axis('off')
+    >>> ax.set_title("Binary test image: foreground in white")
```
Comment (j-bowhay) [inline scipy/ndimage/_morphology.py:1966]: There is no plt.show() here so it shows up with the other figure. That said I am no sure it is necessary to show it twice?

image

Comment (tupui) [reply scipy/ndimage/_morphology.py:1966]: Agreed 👍 The image grid is enough.

File: scipy/ndimage/_morphology.py [hunk 2]
```diff
@@ -1922,12 +1922,93 @@ def distance_transform_bf(input, metric="euclidean", sampling=None,
         Returned only when `return_indices` is True and `indices` is not
         supplied.
 
+    See Also
+    --------
+    distance_transform_cdt : Faster distance transform for taxicab and
+        chessboard metrics
```
Comment (tupui) [inline scipy/ndimage/_morphology.py:1928]:
```suggestion
    distance_transform_cdt : Faster distance transform for taxicab and
                             chessboard metrics
```

File: scipy/ndimage/_morphology.py [hunk 3]
```diff
@@ -1922,12 +1922,93 @@ def distance_transform_bf(input, metric="euclidean", sampling=None,
...
+    >>> fig = plt.figure(figsize=(8, 8))  # set up the figure structure
+    >>> grid = ImageGrid(fig, 111, nrows_ncols=(2, 2), axes_pad=(0.4, 0.3),
+    ...                  label_mode="1", share_all=True,
+    ...                  cbar_location="right", cbar_mode="each",
+    ...                  cbar_size="7%", cbar_pad="2%")
+    >>> for ax in grid:
+    ...     ax.axis('off')  # remove axes from images
+
+    >>> colorbar_ticks = [0, 10, 20]
+    >>> # binary image in top left
```
Comment (tupui) [inline scipy/ndimage/_morphology.py:1982]: Instead of a code block it would be better to separate into sub blocks and make real text comments.

**Review by tupui — COMMENTED** [review]
Thanks! I have a few suggestions.

To the question about tutorial vs docstrings, I am not sure. We just added the infrastructure to add notebooks. In general we need to take some time to rethink the whole organisation. Until it happens, we can still get this in here.

**Comment by j-bowhay** [comment]
A minor point but worth considering: for the plot of the three distance transforms should they use the same limits on the colourbars? This would make it easier to compare the distance between metrics (at the moment it is not immediately obvious that the taxicab metric gives further distances than the others).

**Review by j-bowhay — APPROVED** [review]
Thanks @dschmitz89

**Pull request merged**